client/lib/memoize-last/index.ts

```diff
@@ -6,23 +11,35 @@
  *
  * @returns The wrapped function.
  */
-export default function memoizeLast< T extends ( ...args: any[] ) => any >( fn: T ): T {
-	let lastArgs: Parameters< T >;
-	let lastResult: ReturnType< T >;
+export default function memoizeLast< T extends ( ...args: any[] ) => any >( fn: T ): T & Clearable {
+	let lastArgs: Parameters< T > | symbol = UNSET_SYMBOL;
+	let lastResult: ReturnType< T > | symbol = UNSET_SYMBOL;
+
+	const func = ( ( ...args: Parameters< T > ) => {
+		if ( lastArgs === UNSET_SYMBOL ) {
+			lastArgs = args;
+			lastResult = fn( ...args );
+			return lastResult;
+		}
 
-	return ( ( ...args: Parameters< T > ) => {
 		const isSame =
-			lastArgs &&
-			args.length === lastArgs.length &&
-			args.every( ( arg, index ) => arg === lastArgs[ index ] );
+			args.length === ( lastArgs as Parameters< T > ).length &&
```

Comment:
There should be no need to change this code either. `lastArgs` is either an array (a possibly empty one) or `undefined`. The former means that it hasn't been initialised yet, or has ben cleared; the latter means that it has been initialised.

Reply:
ah, i thought args was `undefined` when not passed, not an empty array. nice.